upstairs/src/upstairs.rs

```diff
@@ -3504,4 +3508,641 @@ pub(crate) mod test {
             assert_eq!(c.state(), DsState::New);
         }
     }
+
+    #[tokio::test]
+    async fn good_decryption() {
```

Comment:
This test is called "good_decryption" but the comment on line 3527 says we
are going to fail decryption?  which is it :)

Reply:
Oops, fixed